util/groupmember/groupmember.go

```diff
@@ -7,14 +7,44 @@
 package groupmember
 
 import (
-	"errors"
-	"runtime"
+	"os/user"
+	"sync"
 )
 
-var ErrNotImplemented = errors.New("not implemented for GOOS=" + runtime.GOOS)
-
 // IsMemberOfGroup reports whether the provided user is a member of
 // the provided system group.
 func IsMemberOfGroup(group, userName string) (bool, error) {
-	return isMemberOfGroup(group, userName)
+	u, err := user.Lookup(userName)
+	if err != nil {
+		return false, err
+	}
+	ugids, err := u.GroupIds()
+	if err != nil {
+		return false, err
+	}
+	gid, err := getGroupID(group)
+	if err != nil {
+		return false, err
+	}
+	for _, ugid := range ugids {
+		if gid == ugid {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+var groupIDCache sync.Map // from string (group name) to string (Gid)
+
+func getGroupID(groupName string) (string, error) {
+	s, ok := groupIDCache.Load(groupName)
```

Comment:
caches for security stuff like this scare me a bit.

also, a more common pattern I think is:

      if s, ok := groupIDCache.Load(groupName).(string); ok {
             return s, nil
       }

That's safe even if somebody puts the wrong value type in that key.

Reply:
But I guess it was already caching, so sure?

Reply:
I can just drop the cache

Reply:
Dropped the cache

Reply:
PTAL

Reply:
Either way. Or a TODO to investigate whether it matters.

In any case, the cache should probably be up a level on the func itself (is this user in this group?) and only cache positive caching for 10 seconds or so.